repository/src/main/java/com/epam/esm/repository/builder/QueryBuilder.java

```diff
@@ -0,0 +1,111 @@
+package com.epam.esm.repository.builder;
+
+import com.epam.esm.repository.SearchCriteria;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import static com.epam.esm.repository.Parameters.*;
+
+@Component
+public final class QueryBuilder {
+    String newString;
+
+    public static class Builder {
+        private final SearchCriteria searchCriteria;
+        private String findBy;
+        private String orderBy;
+        MapSqlParameterSource parameters;
+        String sqlString;
+
+        public Builder(SearchCriteria searchCriteria, String sqlString) {
+            this.sqlString = sqlString;
+            this.searchCriteria = searchCriteria;
+        }
+
+        public Builder buildWhereClause(MapSqlParameterSource parameters) {
+            List<String> list = new ArrayList<>();
+            if (this.searchCriteria.getTagName() != null && !this.searchCriteria.getTagName().isEmpty()) {
+                list.add("tag.name = :" + TAG_NAME_PARAMETER);
+                parameters.addValue(TAG_NAME_PARAMETER, this.searchCriteria.getTagName());
+            }
+
+            if (this.searchCriteria.getCertificateName() != null && !this.searchCriteria.getCertificateName().isEmpty()) {
+                list.add("cert.name = :" + NAME_PARAMETER);
+                parameters.addValue(NAME_PARAMETER, this.searchCriteria.getCertificateName());
+            }
+
+            if (this.searchCriteria.getCertificateDescription() != null && !this.searchCriteria.getCertificateDescription().isEmpty()) {
+                list.add("cert.description = :" + DESCRIPTION_PARAMETER);
+                parameters.addValue(DESCRIPTION_PARAMETER, this.searchCriteria.getCertificateDescription());
+            }
+
+            StringBuilder stringBuilder = new StringBuilder();
+            if (!list.isEmpty()) {
+                stringBuilder.append("WHERE ");
+                Iterator<String> iterator = list.iterator();
```

Comment:
@MakarevichK  Didn't understand how can I replace iterator with stream, how can I check if there is an element after current one, and append to the current if there is?

Reply:
As I understand your logic it can be rewritten as:
`if (!list.isEmpty()) {
                stringBuilder.append("WHERE ");
String expression = list.stream().collect(Collectors.joining(" AND "));
stringBuilder.append(expression);
}`
or even
`if (!list.isEmpty()) {
                stringBuilder.append("WHERE ");
String expression = String.join(" AND ", list);
stringBuilder.append(expression);
}`

Reply:
Didn't get it, just check this logic with 3 elements in list in my code and everything works